### PR TITLE
Term query set

### PR DESCRIPTION
**Pull request by javitonino**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb_node/state.bincode
Comment (hermeGarcia) [inline nucliadb_node/state.bincode:1]: should this be in the PR?

File: nucliadb_paragraphs3/src/reader.rs [hunk 1]
```diff
@@ -132,6 +132,7 @@ impl ParagraphReader for ParagraphReaderService {
     #[measure(actor = "paragraphs", metric = "search")]
     #[tracing::instrument(skip_all)]
     fn search(&self, request: &ProtosRequest, context: &ParagraphsContext) -> NodeResult<ProtosResponse> {
+        println!("{request:#?}");
```
Comment (hermeGarcia) [inline nucliadb_paragraphs3/src/reader.rs:135]: Should this be in the PR?

**Review by hermeGarcia — CHANGES_REQUESTED** [review]
Looks good! Just some things that should be removed before merging

File: nucliadb_vectors/src/formula/mod.rs [hunk 2]
```diff
@@ -18,37 +18,41 @@
...
-        match self.kind {
-            AtomKind::KeyPrefix => retriever.get_key(x).starts_with(self.value.as_bytes()),
-            AtomKind::Label => retriever.has_label(x, self.value.as_bytes()),
+        match self {
+            Self::KeyPrefix(value) => retriever.get_key(x).starts_with(value.as_bytes()),
+
+            Self::Label(value) => retriever.has_label(x, value.as_bytes()),
+            Self::KeyPrefixSet(set) => {
+                let key = retriever.get_key(x);
+                let kk = String::from_utf8_lossy(key);
```
Comment (hermeGarcia) [inline nucliadb_vectors/src/formula/mod.rs:49]: maybe this is not the best name..

File: nucliadb_vectors/src/formula/mod.rs [hunk 3]
```diff
@@ -18,37 +18,41 @@
...
+                let parts: Vec<_> = kk.split('/').collect();
+                if parts.len() < 3 {
+                    return false;
+                }
+                set.contains(&parts[0..3].join("/"))
```
Comment (hermeGarcia) [inline nucliadb_vectors/src/formula/mod.rs:54]: Maybe im wrong, but in line `55` you are splitting the whole string by '/' to then take the first three chunks and joining them again by `/`, wouldn't it be better to do this in a single iteration through the key that builds the string you want?

**Review by hermeGarcia — CHANGES_REQUESTED** [review]

File: nucliadb_vectors/src/formula/mod.rs (hunk 2 above)
Comment (javitonino) [reply nucliadb_vectors/src/formula/mod.rs:49]: :laughing:

File: nucliadb_vectors/src/formula/mod.rs (hunk 3 above)
Comment (javitonino) [reply nucliadb_vectors/src/formula/mod.rs]: Indeed. I just copied this piece of code from the old code at the paragraphs index. I cleaned that one up, but this one remained. I'll make it better.

File: nucliadb_node/state.bincode
Comment (javitonino) [reply nucliadb_node/state.bincode:1]: WTF? Not sure how this ended up outside the data directory :confused: Thanks!

File: nucliadb_paragraphs3/src/reader.rs (hunk 1 above)
Comment (javitonino) [reply nucliadb_paragraphs3/src/reader.rs:135]: :fire:

**Review by hermeGarcia — APPROVED** [review]

**Pull request merged**